utils/timewindow.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"time"
+)
+
+const maintenanceTimeLayout = "150405-0700"
+
+// MaintenanceTime is a structure holding a maintenance time.
+type MaintenanceTime struct {
+	hour   int
+	minute int
+	second int
+}
+
+// NewMaintenanceTime returns a maintenance time structure based on the given hour, minute, and second.
+func NewMaintenanceTime(hour, minute, second int) *MaintenanceTime {
+	if hour >= 24 {
+		panic(fmt.Sprintf("invalid hour %d", hour))
```

Comment:
Is `panic` necessary? Won't returning error do? Some misconfigured CRD should not cause the controller to panic, no?

Reply:
This is not supposed to happen even if user misconfigures the CRD